temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -416,14 +437,13 @@ private AttemptTaskHandlerImpl(ActivityTaskHandler handler) {
 
     @Override
     public void handle(LocalActivityAttemptTask attemptTask) throws Exception {
```

Comment:
I would need to double check the Java SDK, but is `handle` here called per attempt of a local activity? Is there a risk we `markSlotUsed` multiple times for the same activity?

Reply:
Good call. Added a test & fixed.

Reply:
Looking at the original code it seems like before the local activity slot metric was not considered "used" while the local activity was backing off. Now I think the slot will be considered "used" if the activity is executing or not. Do you think I am correct? If so I think that is fine since the slot is not available.

Reply:
Actually looking at the code more it appears there is a more significant change here. Before the local activity concurrency was limited by the number of threads in the thread pool, and more then the max local activity task slot could be scheduled. Now we are limiting the max number of local activities being scheduled to the number of local activity slots. That could have implications to users who use local activities for certain use cases like polling.

Reply:
Discussed on call:

* Will keep the ratelimiter which is 2x max size, or for implementations w/o max size, 2x number of currently running activities (only to be triggered if tryReserve fails).
* Release slots for LAs which are backing off, prioritize retries and re-acquire then